src/HVML/Reduce.hs

```diff
@@ -174,24 +174,36 @@ reduceRefAt book host = do
   let loc = termLoc term
   let fid = lab
   let ari = funArity book fid
-  case fid of
-    x | x == _DUP_F_ -> reduceRefAt_DupF book host loc ari
-    x | x == _SUP_F_ -> reduceRefAt_SupF book host loc ari
-    x | x == _LOG_F_ -> reduceRefAt_LogF book host loc ari
-    oterwise -> case MS.lookup fid (fidToFun book) of
-      Just ((copy, args), core) -> do
-        incItr
-        when (length args /= fromIntegral ari) $ do
-          putStrLn $ "RUNTIME_ERROR: arity mismatch on call to '@" ++ mget (fidToNam book) fid ++ "'."
+  if isValidFid fid then
```

Comment:
isValidFid doesn't do anything right?, because we just fall on the Nothing case below